packages/next/export/index.ts

```diff
@@ -418,12 +418,8 @@ export default async function exportApp(
         })
       )
 
-    if (
-      !options.buildExport &&
-      !exportPathMap['/404'] &&
-      !exportPathMap['/404.html']
-    ) {
-      exportPathMap['/404'] = exportPathMap['/404.html'] = {
+    if (!options.buildExport && !exportPathMap['/404']) {
+      exportPathMap['/404'] = {
         page: '/_error',
       }
     }
```

Comment:
IMHO it is not the correct fix.

The code here should work like this:

- When you specify a custom 404 in `exportPathMap`:
  - Next.js shouldn't overwrite your custom 404 export alias.
- When you don't specify a custom 404 in `exportPathMap`:
  - When `trailingSlash` is disabled (which is by default):
    - only `404.html` should be exported.
    - `['/404', '/404.html']` will be deduped to `/404` after path normalization
  - When `trailingSlash` is enabled:
    - should export `404/index.html` instead.
    - A `404.html` should also be exported for backward compatibility with GitHub Pages, GitLab Pages, Cloudflare Pages, Netlify, etc.